src/main/java/frc/robot/subsystems/VisionSubsystem.java

```diff
@@ -178,6 +179,12 @@ private void process(PhotonPipelineResult result, PhotonPoseEstimator estimator)
     if (estimatedPose.isPresent()) {
       var TimestampSeconds = estimatedPose.get().timestampSeconds;
       var FieldPose3d = estimatedPose.get().estimatedPose;
+      RawfieldPose3dEntry.set(FieldPose3d);
+      if (!MathUtil.isNear(0, FieldPose3d.getZ(), 0.10)
+          || !MathUtil.isNear(0, FieldPose3d.getRotation().getX(), Units.degreesToRadians(3))
+          || MathUtil.isNear(0, FieldPose3d.getRotation().getY(), Units.degreesToRadians(3))) {
```

Comment:
This is worth a comment explaining what this filtering is doing. I'm also not sure why a X rotation larger than 3 degrees is a good thing, but a Y rotation larger than 3 degrees is bad.

Reply:
Nice catch! That's a bug- I guess we forgot the `!` on the last one.